joint_trajectory_controller/doc/userdoc.rst

```diff
@@ -101,6 +113,105 @@ A yaml file for using it could be:
+
+Sending an empty trajectory message from the topic interface (not the action interface) will override the current action goal and not abort the action.
+
+Trajectory replacement [#f2]_
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Joint trajectory messages allow to specify the time at which a new trajectory should start executing by means of the header timestamp, where zero time (the default) means "start now".
+
+The arrival of a new trajectory command does not necessarily mean that the controller will completely discard the currently running trajectory and substitute it with the new one.
+Rather, the controller will take the useful parts of both and combine them appropriately, yielding a smarter trajectory replacement strategy.
+
+The steps followed by the controller for trajectory replacement are as follows:
+
+  + Get useful parts of the new trajectory: Preserve all waypoints whose time to be reached is in the future, and discard those with times in the past.
+    If there are no useful parts (ie. all waypoints are in the past) the new trajectory is rejected and the current one continues execution without changes.
+
+  + Get useful parts of the current trajectory: Preserve the current trajectory up to the start time of the new trajectory, discard the later parts.
+
+  + Combine the useful parts of the current and new trajectories.
+
+The following examples describe this behavior in detail.
+
+The first example shows a joint which is in hold position mode (flat grey line labeled *pos hold* in the figure below).
+A new trajectory (shown in red) arrives at the current time (now), which contains three waypoints and a start time in the future (*traj start*).
+The time at which waypoints should be reached (``time_from_start`` member of ``trajectory_msgs/JointTrajectoryPoint``) is relative to the trajectory start time.
+
+The controller splices the current hold trajectory at time *traj start* and appends the three waypoints.
+Notice that between now and *traj start* the previous position hold is still maintained, as the new trajectory is not supposed to start yet.
+After the last waypoint is reached, its position is held until new commands arrive.
+
+.. image:: new_trajectory.png
+  :alt: Receiving a new trajectory.
+
+|
+
+The controller guarantees that the transition between the current and new trajectories will be smooth. Longer times to reach the first waypoint mean slower transitions.
+
+The next examples discuss the effect of sending the same trajectory to the controller with different start times.
+The scenario is that of a controller executing the trajectory from the previous example (shown in red),
+and receiving a new command (shown in green) with a trajectory start time set to either zero (start now),
+a future time, or a time in the past.
+
+.. image:: trajectory_replacement_future.png
+  :alt: Trajectory start time in the future.
+
+|
+
+.. image:: trajectory_replacement_now.png
```

Comment:
Is this the current behavior? I don't think so: At t=81 and t=89 I sent the action once, at t=100 I sent the action before the first one was finished and it does not interpolate from the old one?
![image](https://user-images.githubusercontent.com/3367244/230723723-0500692d-1808-488d-926d-f61b460d9ac8.png)

tested with the gazebo demos [with a custom action client](https://github.com/christophfroehlich/gazebo_ros2_control/blob/jtc_tests/gazebo_ros2_control_demos/examples/example_position_with_velocity.cpp)

Reply:
This looks strange... maybe we have a bug here. I was debugging few things before, but not sure right now.

Reply:
I removed this section from this PR until we figured out how this really works.

Reply:
see #574